src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -196,32 +196,73 @@ public void GetFullPath_InvalidArgs()
-            const int Iters = 10000;
-            var longPath = new StringBuilder(curDir, curDir.Length + (Iters * 2));
-            for (int i = 0; i < 10000; i++)
+            get
+            {
+                string currentDirectory = Directory.GetCurrentDirectory();
+                string root = Path.GetPathRoot(currentDirectory);
+                string fileName = Path.GetFileName(currentDirectory);
+
+                TheoryData<string, string> data = new TheoryData<string, string>
+                {
+                    // Current directory => current directory
+                    { currentDirectory, currentDirectory },
+                    // "." => current directory
+                    { ".", currentDirectory },
+                    // ".." => up a directory
+                    { "..", Path.GetDirectoryName(currentDirectory) },
+                    // "dir/./././." => "dir"
+                    { Path.Combine(currentDirectory, ".", ".", ".", ".", "."), currentDirectory },
+                    // "dir///." => "dir"
+                    { currentDirectory + new string(Path.DirectorySeparatorChar, 3) + ".", currentDirectory },
+                    // "dir/../dir/./../dir" => "dir"
+                    { Path.Combine(currentDirectory, "..", fileName, ".", "..", fileName), currentDirectory },
+                    // "C:\somedir\.." => "C:\"
+                    { Path.Combine(root, "somedir", ".."), root },
+                    // "C:\." => "C:\"
+                    { Path.Combine(root, "."), root },
+                    // "C:\..\..\..\.." => "C:\"
+                    { Path.Combine(root, "..", "..", "..", ".."), root },
+                    // "C:\\\" => "C:\"
+                    { root + new string(Path.DirectorySeparatorChar, 3), root },
+                };
+
+                // Path longer than MaxPath that normalizes down to less than MaxPath
+                const int Iters = 10000;
+                var longPath = new StringBuilder(currentDirectory, currentDirectory.Length + (Iters * 2));
+                for (int i = 0; i < 10000; i++)
+                {
+                    longPath.Append(Path.DirectorySeparatorChar).Append('.');
+                }
+                data.Add(longPath.ToString(), currentDirectory);
+
+                return data;
+            }
+        }
+
+        public static TheoryData<string, string> GetFullPath_TildePaths
+        {
+            get
             {
-                longPath.Append(Path.DirectorySeparatorChar).Append('.');
+                // Paths with tildes '~' are processed for 8.3 expansion on Windows
+                string currentDirectory = Directory.GetCurrentDirectory();
+                string root = Path.GetPathRoot(currentDirectory);
+
+                TheoryData<string, string> data = new TheoryData<string, string>
+                {
+                    { "~", Path.Combine(currentDirectory, "~") },
```

Comment:
So GetFullPath("~") results in %CD%\~ ? How does that relate to `Paths with tildes '~' are processed for 8.3 expansion on Windows`. Are you just validating that `~` is not special (anymore?)

Reply:
comment maybe?

Reply:
Anything with a tilde gets passed to 8.3 expansion. We're just passing "normal" paths through that hit that code path to look for fall-on-our-face bugs. A file can be named `"~"` (although I wouldn't recommend it).